module/motion/QuinticWalk/src/engine/TrajectoryUtils.hpp

```diff
@@ -0,0 +1,308 @@
+                    case Value::TRUNK_POS_Z: return "TRUNK_POS_Z";
+                    case Value::TRUNK_AXIS_X: return "TRUNK_AXIS_X";
+                    case Value::TRUNK_AXIS_Y: return "TRUNK_AXIS_Y";
+                    case Value::TRUNK_AXIS_Z: return "TRUNK_AXIS_Z";
+                    case Value::FOOT_POS_X: return "FOOT_POS_X";
+                    case Value::FOOT_POS_Y: return "FOOT_POS_Y";
+                    case Value::FOOT_POS_Z: return "FOOT_POS_Z";
+                    case Value::FOOT_AXIS_X: return "FOOT_AXIS_X";
+                    case Value::FOOT_AXIS_Y: return "FOOT_AXIS_Y";
+                    case Value::FOOT_AXIS_Z: return "FOOT_AXIS_Z";
+                    default:
+                        throw std::runtime_error("enum TrajectoryTypes's value is corrupt, unknown value stored"
+                                                 + std::to_string(static_cast<uint8_t>(value)));
+                }
+            }
+        };
+
+        /**
+         * Simple typedef for trajectories container
+         */
+        using Trajectories = SplineContainer<SmoothSpline<float>, TrajectoryTypes, float>;
+
+        /**
+         * Return initialized trajectories for trunk/foot ik cartesian with empty splines
+         */
+        inline void TrajectoriesInit(Trajectories& traj) {
+            if (traj.size() != 0) {
+                traj.reset();
+            }
+            else {
+                traj.add(TrajectoryTypes::IS_DOUBLE_SUPPORT);
+                traj.add(TrajectoryTypes::IS_LEFT_SUPPORT_FOOT);
+                traj.add(TrajectoryTypes::TRUNK_POS_X);
+                traj.add(TrajectoryTypes::TRUNK_POS_Y);
+                traj.add(TrajectoryTypes::TRUNK_POS_Z);
+                traj.add(TrajectoryTypes::TRUNK_AXIS_X);
+                traj.add(TrajectoryTypes::TRUNK_AXIS_Y);
+                traj.add(TrajectoryTypes::TRUNK_AXIS_Z);
+                traj.add(TrajectoryTypes::FOOT_POS_X);
+                traj.add(TrajectoryTypes::FOOT_POS_Y);
+                traj.add(TrajectoryTypes::FOOT_POS_Z);
+                traj.add(TrajectoryTypes::FOOT_AXIS_X);
+                traj.add(TrajectoryTypes::FOOT_AXIS_Y);
+                traj.add(TrajectoryTypes::FOOT_AXIS_Z);
+            }
+        }
+
+        /**
+         * Compute from given spline container trajectory Cartesian trunk and foot position/velocity/acceleration and
+         * assign it to given vector
+         */
+        inline void TrajectoriesTrunkFootPos(float t,
+                                             const Trajectories& traj,
+                                             Eigen::Vector3f& trunkPos,
+                                             Eigen::Vector3f& trunkAxis,
+                                             Eigen::Vector3f& footPos,
+                                             Eigen::Vector3f& footAxis) {
+            if (traj.size() != 14) {
+                NUClear::log(traj.size());
```

Comment:
Should probably make this say something more descriptive, and maybe make it a warning?